addons/base_report_engine_paper_muncher/engine/http.py

```diff
@@ -0,0 +1,113 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import io
+
+from http import HTTPStatus
+
+MAX_BUFFER_SIZE = 2**10
+
+
+class HttpMessage:
+
+    def __init__(self):
+        self.headers = {}
+
+    def _readHeaderLines(self, reader: io.BufferedReader) -> list[str]:
+        lines = []
+
+        while True:
+            request_line = reader.readline().decode('utf-8')
+
+            if len(request_line) == 0:
+                raise EOFError("Input stream has ended")
+
+            if request_line == "\r\n":
+                break
+
+            lines.append(request_line)
+
+        return lines
+
+    def _addToHeader(self, header_line: str) -> None:
+        key, value = header_line.split(':')
+        self.headers[key.strip()] = value.strip()
+
+    def readHeader(self, reader: io.TextIOWrapper) -> None:
+        raise NotImplementedError()
+
+    def _readSingleChunk(self, reader: io.TextIOWrapper) -> bytes:
+        def read_chunk_content(rem_size):
+            chunk = b""
+
+            while rem_size > 0:
+                bs = min(MAX_BUFFER_SIZE, rem_size)
+                byte = reader.read(bs)
+                chunk += byte
+
+                rem_size -= bs
+            return chunk
+
+        size = int(reader.readline()[:-2])
+        chunk = read_chunk_content(size)
+
+        reader.read(2)
```

Comment:
Add a comment that you want to cosume '\r\n' (and actually test that it is what is read here).